db/db_compaction_test.cc

```diff
@@ -6278,248 +6273,534 @@ void IngestOneKeyValue(DBImpl* db, const std::string& key,
+            int64_t* current_time = (int64_t*)current_time_arg;
+            *current_time = *current_time +
+                            options_.compaction_options_fifo.age_for_warm + 1;
+          });
+      ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
+          "PostPickCompactionToWarm", [&](void* compaction_arg) {
+            Compaction* compaction = (Compaction*)compaction_arg;
+            if (compaction != nullptr) {
+              compaction_path_sync_point_called_.store(true);
+            }
+          });
+    }
 
-  ASSERT_OK(Flush());
-}
+    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->EnableProcessing();
+  }
 
-TEST_P(DBCompactionTestWithParam,
-       IntraL0CompactionAfterFlushCheckConsistencyFail) {
-  Options options = CurrentOptions();
-  options.force_consistency_checks = true;
-  options.compression = kNoCompression;
-  options.level0_file_num_compaction_trigger = 5;
-  options.max_background_compactions = 2;
-  options.max_subcompactions = max_subcompactions_;
-  options.write_buffer_size = 2 << 20;
-  options.max_write_buffer_number = 6;
-  DestroyAndReopen(options);
+  bool SyncPointsCalled() { return compaction_path_sync_point_called_.load(); }
 
-  const size_t kValueSize = 1 << 20;
-  Random rnd(301);
-  std::string value(rnd.RandomString(kValueSize));
-  std::string value2(rnd.RandomString(kValueSize));
-  std::string bigvalue = value + value;
+  void DisableSyncPoints() {
+    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->ClearAllCallBacks();
+    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
+  }
+
+  // For files of largest seqno less than `strict_largest_seqno_uppper_bound`,
+  // return the largest seqno of the latest L0 file (based on file number)
+  SequenceNumber GetLatestL0FileLargestSeqnoHelper(
+      SequenceNumber strict_largest_seqno_uppper_bound = kMaxSequenceNumber) {
+    VersionSet* const versions = dbfull()->GetVersionSet();
+    assert(versions);
+    ColumnFamilyData* const cfd = versions->GetColumnFamilySet()->GetDefault();
+    assert(cfd);
+    Version* const current = cfd->current();
+    assert(current);
+    VersionStorageInfo* const storage_info = current->storage_info();
+    assert(storage_info);
+    const std::vector<FileMetaData*> level0_files = storage_info->LevelFiles(0);
+    assert(level0_files.size() >= 1);
+
+    uint64_t latest_file_num = 0;
+    uint64_t latest_file_largest_seqno = 0;
+    for (FileMetaData* f : level0_files) {
+      if (f->fd.largest_seqno < strict_largest_seqno_uppper_bound &&
+          f->fd.GetNumber() > latest_file_num) {
+        latest_file_num = f->fd.GetNumber();
+        latest_file_largest_seqno = f->fd.largest_seqno;
+      }
+    }
+
+    return latest_file_largest_seqno;
+  }
+
+ protected:
+  Options options_;
+
+ private:
+  const Snapshot* snapshot_ = nullptr;
+  std::atomic<bool> compaction_path_sync_point_called_;
+};
+
+TEST_F(DBCompactionTestL0FilesReorderingCorruption,
```

Comment:
I decided to replace original `DBCompactionTestWithParam.IntraL0CompactionAfterFlushCheckConsistencyFail` and `DBCompactionTestFIFOCheckConsistencyWithParam.FlushAfterIntraL0CompactionWithIngestedFile` with one test case here cuz differentiating by memtable being sealed or not does not seem quite important for L0 epoch number approach.